File: scanport.py

print("\nDeveloped By: Junior Mario and Diego Bernardes")
import socket as S
ip = input("Digite o IP a ser scaneado: ")
portas = input("Digite a quantidade de portas a serem scaneadas: ")

if ',' in portas: portas = portas.split(',')
elif '-' == portas: portas = range(1,65536)
elif '-' in portas:
        portas = portas.split('-')
        portas = range(int(portas[0]), int(portas[1])+1)
else: portas = [int(portas)]

socket = S.socket(S.AF_INET, S.SOCK_STREAM)
def analisa(ip, port):#faz a tentativa de conexão com a porta
	try:
		socket.connect((ip, port))
		return True
	except:
		return False
for port in portas:
        resposta = analisa(ip, int(port))
        if resposta == True:
                print('porta {port} ABERTA em {ip}'.format(port = port, ip = ip))
        else:
                print('porta {port} FECHADA em {ip}'.format(port = port, ip = ip))

File: scanport.rb
require 'socket'
def connect(ip, port)
	begin
		con = TCPSocket.open(ip, port)
		return true
	rescue Errno::ECONNREFUSED
		return false
	end
end
STDOUT.print '[IP a ser Scaneado]:' 
ip = gets.chomp
STDOUT.print '[Digite as portas a serem Scaneadas]:'
porta = gets.chomp
if porta.include? '-'
	porta = porta.split('-')
	portas = porta[0].to_i...porta[1].to_i
	puts portas
elsif porta.include? ","
	arra = []
	porta = porta.split(',')
	for ara in porta
		arra << ara 	
	end
	portas = arra
else
	puts "Use - para percorrer as portas e , para portas especificas"
end
for port in portas
	answer = connect(ip, port)
	if answer == true
		puts "Porta #{port} aberta em #{ip}"
	elsif answer == false
		puts "Porta #{port} fechada em #{ip}"
	else
		puts "Nada esta sendo retornado. :("
	end
end